cmd/dagger/cmd/project/template.go

```diff
@@ -0,0 +1,50 @@
+package project
+
+import (
+	"embed"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+//go:embed templates/*.cue
+var content embed.FS
+
+func createTemplate(name string) error {
+	filename := fmt.Sprintf("%s.cue", name)
+	f, err := content.Open(fmt.Sprintf("templates/%s", filename))
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	fout, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+
+	defer fout.Close()
+
+	_, err = io.Copy(fout, f)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getTemplateNames() ([]string, error) {
+	r := []string{}
```

Comment:
I've seen this differently in multiple codebases (i.e: https://cs.opensource.google/go/go/+/master:src/os/dir.go;l=75?q=%22%5B%5Dstring%7B%7D%22&ss=go%2Fgo&start=31) 

The main reason here to return `[]string{}` is that otherwise, I'll have to either initialize it below even for empty results as we shouldn't return a `nil` value here.